test/test-TabTracker.js

```diff
@@ -16,6 +16,37 @@ const EXPECTED_KEYS = ["url", "tab_id", "session_duration", "client_id", "unload
 let ACTIVITY_STREAMS_URL;
 let app;
 
+function createPingSentPromise(pingData, expectedPingCount) {
+  return new Promise(resolve => {
+    let Observer = {
+      observe: function(subject, topic, data) {
```

Comment:
nit: the code can be made simpler by just passing a named function to the observer service.

note that it has to be a named function:

```javascript
function someFunc(subject, topic, data) {
  ...
  Services.obs.removeObserver(someFunc, someNotif);
}
Services.obs.addObserver(someFunc, someNotif);